src/index.js

```diff
@@ -209,6 +216,10 @@ export class BrowserHistory extends History {
       this.options.routeHandler(fragment) :
       false;
   }
+
+  _createOrigin(protocol: string, hostname: string, port: string) {
```

Comment:
Sounds good. For a unit test I think it'd also be fine to simply overwrite `bs.location` with an object containing the values used in the specific test case.